### ISSUE TITLE
在i=1的时候，没有进行分解的时候，好像loss的结果和输出的checkpoint test结果不太能对的上

### ISSUE THREAD

**Issue by ggrrww** [wengwenchao123/DDGCRN#13]
这是我的log文件，训到最后average loss已经降到14.40了，但是在checkpoint测试的时候又突然变得很大，不知道是怎么回事

2025-11-17 16:05: ********Train Epoch 98: averaged Loss: 13.491078, train time: 18.44 s
2025-11-17 16:05: ***********Val Epoch 98: average Loss: 14.632330, train time: 1.81 s
2025-11-17 16:05: **********test Epoch 98: average Loss: 14.401498, train time: 1.59 s
2025-11-17 16:05: Validation performance didn't improve for 15 epochs. Training stops.
2025-11-17 16:05: Saving current best model to /root/code/DDGCRN/experiments/PEMSD8/20251117153022/best_model.pth
2025-11-17 16:05: Saving current best model to /root/code/DDGCRN/experiments/PEMSD8/20251117153022/best_test_model.pth
2025-11-17 16:05: Horizon 01, MAE: 30.0500, RMSE: 37.0587, MAPE: 0.3933
2025-11-17 16:05: Horizon 02, MAE: 52.1174, RMSE: 57.7499, MAPE: 0.6637
2025-11-17 16:05: Horizon 03, MAE: 47.2215, RMSE: 53.4988, MAPE: 0.5882
2025-11-17 16:05: Horizon 04, MAE: 55.2650, RMSE: 61.1360, MAPE: 0.6973
2025-11-17 16:05: Horizon 05, MAE: 22.5212, RMSE: 32.5435, MAPE: 0.1825
2025-11-17 16:05: Horizon 06, MAE: 42.8631, RMSE: 50.2583, MAPE: 0.5047
2025-11-17 16:05: Horizon 07, MAE: 31.8966, RMSE: 40.9873, MAPE: 0.3268
2025-11-17 16:05: Horizon 08, MAE: 91.5434, RMSE: 96.1703, MAPE: 1.1698
2025-11-17 16:05: Horizon 09, MAE: 29.0503, RMSE: 37.3258, MAPE: 0.3954
2025-11-17 16:05: Horizon 10, MAE: 25.2297, RMSE: 34.4800, MAPE: 0.3051
2025-11-17 16:05: Horizon 11, MAE: 51.6940, RMSE: 59.9185, MAPE: 0.5761
2025-11-17 16:05: Horizon 12, MAE: 84.0430, RMSE: 90.1904, MAPE: 1.0224
2025-11-17 16:05: Average Horizon, MAE: 46.9579, RMSE: 57.8276, MAPE: 0.5688
2025-11-17 16:05: This is best_test_model
2025-11-17 16:05: Horizon 01, MAE: 30.0500, RMSE: 37.0587, MAPE: 0.3933
2025-11-17 16:05: Horizon 02, MAE: 52.1174, RMSE: 57.7499, MAPE: 0.6637
2025-11-17 16:05: Horizon 03, MAE: 47.2215, RMSE: 53.4988, MAPE: 0.5882
2025-11-17 16:05: Horizon 04, MAE: 55.2650, RMSE: 61.1360, MAPE: 0.6973
2025-11-17 16:05: Horizon 05, MAE: 22.5212, RMSE: 32.5435, MAPE: 0.1825
2025-11-17 16:05: Horizon 06, MAE: 42.8631, RMSE: 50.2583, MAPE: 0.5047
2025-11-17 16:05: Horizon 07, MAE: 31.8966, RMSE: 40.9873, MAPE: 0.3268
2025-11-17 16:05: Horizon 08, MAE: 91.5434, RMSE: 96.1703, MAPE: 1.1698
2025-11-17 16:05: Horizon 09, MAE: 29.0503, RMSE: 37.3258, MAPE: 0.3954
2025-11-17 16:05: Horizon 10, MAE: 25.2297, RMSE: 34.4800, MAPE: 0.3051
2025-11-17 16:05: Horizon 11, MAE: 51.6940, RMSE: 59.9185, MAPE: 0.5761
2025-11-17 16:05: Horizon 12, MAE: 84.0430, RMSE: 90.1904, MAPE: 1.0224
2025-11-17 16:05: Average Horizon, MAE: 46.9579, RMSE: 57.8276, MAPE: 0.5688

**Comment by wengwenchao123** [comment]
额，你是说你训练的时候只用了一层是吗，然后是通过i=1来实现的。
如果是这个问题，是因为test的时候，我没有从外部输入i的数值，所以模型直接用forward里面默认的i=2来运行的
只用一层的话，我个人建议是直接内部确定i=1或者只保留一层的部分来训练，这样子不会有错

**Comment by ggrrww** [comment]
def forward(self, source, i=1):
我是这边forward设置的i=1，然后出现了这个错误，但是用这个checkpoint mode=test的时候如果设置forward里i=2的时候又正常了

但是我回看了一下源代码，BasicTrainer.py种好像默认设置的也是：train_epoch_loss = self.train_epoch(epoch, 1)，然后传了一个i=1进去，感觉这边逻辑好像有点奇怪

**Comment by ggrrww** [comment]
我好像明白哪里出问题了，因为在BasicTrainer.train那边也传了个i=2进去，所以直接在forward里面改训练的时候就没有效果，i还是等于2，但是直接test的时候，BasicTrainer里面没有传参，就导致i=1，然后两边就没对上导致结果出问题了

**Comment by wengwenchao123** [comment]
大概率就是你说的问题
